tests/main/user-session-env/task.yaml

```diff
@@ -0,0 +1,88 @@
+summary: verify that user environment settings are added
+
+description: |
+    User environment variables are added via /etc/profile.d/snapd.sh (bash/sh
+    specific) or via /lib/environment.d/ helpers. Make sure that at least one of
+    the mechanisms works and XDG_DATA_DIRS and PATH are updated accordingly
+    inside the user session, no matter the shell they use.
+
+# environment generators were introduced in systemd 233, anything before that
+# will not work
+systems:
+   # cannot install zsh
+   - -ubuntu-core-*
+
+environment:
+    TEST_ZSH_USER: test-zsh
+    NO_USER_SESSION_SYSTEMS: ubuntu-14.04 amazon-linux centos-7 debian-9 ubuntu-16.04
```

Comment:
do you know if there is any way to detect if we are running in one of these systems instead of harcoading those? So then if we add a new system or something changes we don't need to update the test

Reply:
AFAIU newly added systems should/will support user session, so the list should not change much/at all.

Reply:
I wonder if there's something simple, such as checking for `$XDR_RUNTIME_DIR/systemd/private`? Having said that I would not cry if we didn't do this.

Reply:
Hm that would exist only once the session is started though?